x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/predicate/operator/comparison/In.java

```diff
@@ -105,6 +108,26 @@ protected Pipe makePipe() {
         return new InPipe(source(), this, children().stream().map(Expressions::pipe).collect(Collectors.toList()));
     }
 
+    @Override
+    protected TypeResolution resolveType() {
+        if (value instanceof FieldAttribute) {
+            try {
+                ((FieldAttribute) value).exactAttribute();
+            } catch (MappingException ex) {
+                return new TypeResolution(format(null, "[{}] cannot operate on first argument field of data type [{}]",
```

Comment:
I am not sure this is the correct error message here. Looking at the test [here](https://github.com/elastic/elasticsearch/pull/38440/files#diff-334bd2f8c33b60789817181fcef4ca52R413) I think it will be confusing for users as to why they get that error message. Meaning, `text` is a text field, and the query is `WHERE text IN ('foo', 'bar')` (everything is text) but the error message says one cannot use text. I think it would help users if the error message would be a bit special for `text`: maybe say between brackets "needs to be of `keyword` type" or something similar.

Reply:
I used the same message in First/Last.
The `MappingException` is thrown in this case (`text` with no underlying `keyword`) but since it's a generic exception I was "afraid" to add a specific message for the case. (More like future proof driven).
@costin what do you think?